packages/react-draggable-dialog/src/components/DraggableDialogSurface/useDraggableDialogSurface.ts

```diff
@@ -109,19 +119,14 @@ export const useDraggableDialogSurface = (
 
     if (!hasBeenDragged) {
       if (!boundaryRect) {
-        return;
+        return undefined;
       }
 
       const boundaryTopReference = boundaryRect.top + boundaryRect.height / 2;
       const boundaryLeftReference = boundaryRect.left + boundaryRect.width / 2;
       const top = boundaryTopReference - Math.ceil(currentEl.clientHeight / 2);
       const left = boundaryLeftReference - Math.ceil(currentEl.clientWidth / 2);
 
-      setDropPosition({
```

Comment:
This was causing React to set the state wrongly, especially after removing the element from DOM